sdk/src/assertions/timestamp.rs

```diff
@@ -20,118 +20,144 @@ use serde_bytes::ByteBuf;
+    /// Refresh the timstamp token for a given manifest id.
+    ///
+    /// The signature is expected to be the `signature` field of the `COSE_Sign1_Tagged` structure
+    /// found in the C2PA claim signature box of the manifest corresponding to the `manifest_id`.
+    //
+    // The `signature` is normally obtained from [`Store::get_cose_sign1_signature`].
+    //
+    // [`Store::get_cose_sign1_signature`][crate::store::Store::get_cose_sign1_structure].
+    #[async_generic(async_signature(
+        &mut self,
+        time_authority_url: &str,
+        manifest_id: &str,
+        signature: &[u8],
+        http_resolver: &impl AsyncHttpResolver,
+    ))]
+    pub(crate) fn refresh_timestamp(
+        &mut self,
+        time_authority_url: &str,
+        manifest_id: &str,
+        signature: &[u8],
+        http_resolver: &impl SyncHttpResolver,
+    ) -> Result<()> {
+        let timestamp_token = if _sync {
+            TimeStamp::send_timestamp_token_request(time_authority_url, signature, http_resolver)?
         } else {
-            Self::send_timestamp_token_request_impl_async(
-                tsa_url,
-                message,
-                &AsyncGenericResolver::new(),
+            TimeStamp::send_timestamp_token_request_async(
+                time_authority_url,
+                signature,
+                http_resolver,
             )
-            .await
-        }
+            .await?
+        };
+
+        self.0
+            .insert(manifest_id.to_owned(), ByteBuf::from(timestamp_token));
+
+        Ok(())
     }
 
+    /// Send a timestamp token request to the `time_authority_url` with the given `signature`.
+    ///
+    /// This function will verify the structure of the returned response but not the trust.
+    ///
+    /// See [`TimeStamp::refresh_timestamp`] for more information.
     #[async_generic(async_signature(
-        tsa_url: &str,
-        message: &[u8],
+        time_authority_url: &str,
+        signature: &[u8],
         http_resolver: &impl AsyncHttpResolver,
     ))]
-    pub(crate) fn send_timestamp_token_request_impl(
-        tsa_url: &str,
-        message: &[u8],
+    pub(crate) fn send_timestamp_token_request(
+        time_authority_url: &str,
+        signature: &[u8],
         http_resolver: &impl SyncHttpResolver,
     ) -> Result<Vec<u8>> {
-        use crate::{
-            crypto::cose::CertificateTrustPolicy, settings::Settings,
-            status_tracker::StatusTracker, Error,
-        };
-
-        let body = crate::crypto::time_stamp::default_rfc3161_message(message)?;
+        let body = crate::crypto::time_stamp::default_rfc3161_message(signature)?;
         let headers = None;
 
         let bytes = if _sync {
             crate::crypto::time_stamp::default_rfc3161_request(
-                tsa_url,
+                time_authority_url,
                 headers,
                 &body,
-                message,
+                signature,
```

Comment:
why did you name this signature?  It may be in this spec but is has changed before.  It is the message payload to be signed.  It is not acutally a signature.